assembly/src/assembler/instruction/u32_ops.rs

```diff
@@ -257,9 +257,7 @@ pub fn u32shl(
     match imm {
         Some(imm) => match op_mode {
             U32OpMode::Checked => {
-                if imm > 31 {
-                    return Err(AssemblerError::imm_out_of_bounds(imm as u64, 0, 31));
-                }
+                validate_param(imm, 0, 31)?;
```

Comment:
non-blocking & unrelated to this PR: it might be worthy to add a comment on why we use `31` since it's non-obvious given the boundary of the type (32)

Maybe even create a documented constant for it

Reply:
N/A, it's already created in the next PR